File: app/src/main/java/com/planet/myapptmdb/model/storage/DataSession.java
package com.planet.myapptmdb.model.storage;

import com.planet.myapptmdb.model.ResultsItem;

import java.util.LinkedList;
import java.util.List;

/**
 * Clase que almacena datos persistentes.
 */

public class DataSession {

    private final static String TAG = DataSession.class.getSimpleName();
    private static DataSession ourInstance = null;

    private List<ResultsItem> moviesSelected;

    private DataSession() {
        moviesSelected = new LinkedList<>();
    }

    public static DataSession getInstance() {
        if (ourInstance == null)
            ourInstance = new DataSession();
        return ourInstance;
    }

    public List<ResultsItem> getMoviesSelected() {
        return moviesSelected;
    }

    public void setMoviesSelected(List<ResultsItem> moviesSelected) {
        this.moviesSelected = moviesSelected;
    }
}

File: app/src/main/java/com/planet/myapptmdb/utils/DataSession.kt
package com.planet.myapptmdb.utils

import com.planet.myapptmdb.model.ResultsItem

import java.util.LinkedList

/**
 * Clase que almacena datos persistentes.
 */

class DataSession private constructor() {

    var moviesSelected: MutableList<ResultsItem>? = null

    init {
        moviesSelected = LinkedList()
    }

    companion object {

        private val TAG = DataSession::class.java.simpleName
        private var ourInstance: DataSession? = null

        val instance: DataSession
            get() {
                if (ourInstance == null)
                    ourInstance =
                        DataSession()
                return ourInstance as DataSession
            }
    }
}
